### PR TITLE
Feature/enhance application detail endpoint

### PR DESCRIPTION
**Pull request by richtier**
Also change the application-level field names to something more readable. database column names remain unchanged

### CODE REVIEW

File: spire/views.py [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
+class LicenceModelView(viewsets.ReadOnlyModelViewSet):
+    permission_classes = [SignatureCheckPermission]
+    authentication_classes = []
+    serializer_class = serializers.LicenceSerializer
+    filterset_class = filters.LicenceFilterSet
+    queryset = (
+        models.Licence.objects
+        .order_by('id')
+        .exclude(licence_status='DRAFT')
+        .prefetch_related(
```
Comment (richtier) [inline spire/views.py:18]: optimized to 7 db reads in per request. we can quite easily add caching as the database hardly ever changes. we can cache using the querystring as key and invalidate it all whenever the db gets updated

File: spire/serializers.py [hunk 2]
```diff
@@ -16,17 +32,40 @@ class Meta(object):
...
-        # licence_detail -> application_detail -> application -> control list
-        return ControlListGoodsSerializer(obj.ela_detail.ela.controllistgoods_set.all(), many=True).data
+        qs = obj.application_detail.application.control_list_good_set.all()
+        return ControlListGoodSerializer(qs, many=True).data
+
+
+class LicenceSerializer(serializers.ModelSerializer):
+    licence_detail = serializers.SerializerMethodField()
+
+    class Meta(object):
```
Comment (saruniitr) [inline spire/serializers.py:55]: Assuming inheriting from `object` is to keep it consistent with existing usage in this file?

File: spire/models.py [hunk 3]
```diff
@@ -271,10 +276,22 @@ class Meta:
 
 class LicenceDetail(models.Model):
     # id = models.ForeignKey('Uref', models.DO_NOTHING, db_column='id', primary_key=True)
-    l = models.ForeignKey(Licence, models.DO_NOTHING, verbose_name='Licence')  # NOQA
+    licence = models.ForeignKey(
+        Licence,
+        db_column='l_id',
```
[the remark this answers is not in the document]
Comment (richtier) [reply spire/models.py:281]: agreed but I was planning on only renaming fields we interact (we use `licence` for example in filters, views, and serializers. we dont use licence_id anywhere yet).

File: spire/serializers.py (hunk 2 above)
Comment (richtier) [reply spire/serializers.py:55]: oops. got overzelous copy and pasting. will remove :+1:

File: spire/models.py [hunk 4]
```diff
@@ -298,7 +315,13 @@ class Meta:
 
 
 class LicenceLine(models.Model):
```
[the remark this answers is not in the document]
Comment (richtier) [reply spire/models.py:317]: agreed the names are not great. we have a document for understanding the schema we've inherited. I'll send it to you

**Pull request merged**